src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -115,6 +115,29 @@ async def _git_fetch(directory: str):
     await run_cmd_line(cmd, f"{directory}")
 
 
+async def _git_get_latest_matching_tag_capture_groups(
+    directory: str, regexp: str
+) -> Optional[tuple[str]]:  # pylint: disable=unsubscriptable-object
+    cmd = [
+        "git",
+        "tag",
+        "--list",
+        "--sort=creatordate",  # Sorted ascending by date
+    ]
+    all_tags = await run_cmd_line(cmd, f"{directory}")
+    if all_tags == None:
+        return None
+    all_tags = all_tags.split("\n")
+    all_tags = [tag for tag in all_tags if tag != ""]
+    list_tags = [tag for tag in all_tags if re.search(regexp, tag) != None]
+    if not list_tags:
+        return None
+    if re.compile(regexp).groups == 0:
```

Comment:
what do you intend to do here? Are you sure this is right?
Typically ``re.compile`` is done to create a regex instance where you call ``match`` or ``search`` and only then you call ``groups`` to see the results

```python

tag_re = re.compile(regexp)

list_tags = [tag for tag in all_tags if tag_re.search(tag) ]
```

Reply:
In this case I want to know if there are capture groups, so literally something like `(.*)` inside the regex. If so, only the first capture group is used.

thanks!